### DISCUSSION TITLE
Enable Github Wiki for winutil

### DISCUSSION THREAD

**Discussion by Marterich** [ChrisTitusTech/winutil#2008] [Ideas]
Would it be smart to enable the GitHub Wiki feature for winutil? This would allow us to keep the Readme short and focused on "What is winutil and how to run it." Additional information like workarounds, alternative start methods, contribution guidelines, and solutions to common problems could be documented in the Wiki, avoiding clutter on the main page and improving the first impression of the tool.

**Comment by og-mrk** [comment]
I did notice this when I first contributed to the project 2-3 months ago, thought Chris simply didn't want GitHub wiki (even though there's a wiki folder, and he (Chris) did talk in one of his livestream about enabling/writing docs using GitHub's Wiki feature).

And I think pointing new users/contributors to a well written source of documentation, like the project's wiki, would be quite nice 👍

**Reply by og-mrk** [reply]
Please note @Marterich that simply enabling wiki doesn't solve this current problem of lack of documentation the project has, I think (if possible of course) that splitting the load of completing this task to multiple core contributors would help alot in speeding things up, and result in better documentation, because even though, lets say Contributor X, has contributed to the project in the last month or two, _or even a year_.. that doesn't mean they understand every aspect of how the project works, as WinUtil is quite a big project (even thought its Lines of Code isn't that much.. it's still a lot to fully understand).. with that being said, anyone who understand any part of the project is invited to go ahead and contribute the docs, as it'll help others understand said part.

**Reply by Marterich** [reply]
I'm fully aware of the implications of saying "We need documentation" from my professional experience, but at the moment my main gripe with this project is, that if there isn't even the option to document in a centralized, standardized way, event the Contributors that would be openminded to write easy to understand documentation simply won't. 

From my personal experience, I would have appreciated something like a "Getting Started" to the inner workings of winutil because it would greatly reduce the time needed to get up to speed on where to implement a change if you are not fully familiar with the codebase yet. 

At the moment I'm playing around with implementing themes that can be changed via a button press at runtime and it once again became painfully clear to me, that even though a project is open source, one can look at the source and reverse engineer how everything works, it can be really time intensive to get up to speed without solid documentation.

**Reply by og-mrk** [reply]
True @Marterich, the drag/friction you've talked about with the "Getting started", lets say with part of the code base, this happened to me every time I wanted to contribute on WinUtil's UX/UI side (I have abit of experience on this doing Game Dev).. but I can't do so without alot of 3d-party documentation, as the UI code doesn't make a lot of sense, it's written in XAML, using WPF for rendering (a .NET Framework by Microsoft), and everything is.. just confusing 😞

**Reply by og-mrk** [reply]
We'll be using some Site Generation with Markdown Files (or a combination of Markdown-like format + html/css), so GitHub Wiki won't be enabled because we already have a place to write documentation into.

@Marterich Can you close this discussing as completed?
